src/app/commands/cmd_export_sprite_sheet.cpp

```diff
@@ -212,13 +219,16 @@ Doc* generate_sprite_sheet_from_params(
   // If the user choose to render selected layers only, we can
   // temporaly make them visible and hide the other ones.
   RestoreVisibleLayers layersVisibility;
-  calculate_visible_layers(site, layerName, layersVisibility);
+  calculate_visible_layers(site, layerName, layerIndex, layersVisibility);
 
   SelectedLayers selLayers;
   if (layerName != kSelectedLayers) {
     // TODO add a getLayerByName
+    int i = sprite->allLayersCount();
     for (const Layer* layer : sprite->allLayers()) {
-      if (layer->name() == layerName) {
+      i--;
+      if (layer->name() == layerName && layerIndex == -1 ||
+          layer->name() == layerName && layerIndex == i) {
```

Comment:
@martincapello just saw a warning (`warning: suggest parentheses around ‘&&’ within ‘||’ [-Wparentheses]`) about all these conditions, it's already merged but it would be nice to fix it:
```c++
      if (layer->name() == layerName && layerIndex == -1 ||
          layer->name() == layerName && layerIndex == i) {
```
It's equal to:
```c++
      if ((layer->name() == layerName) && (layerIndex == -1 || layerIndex == i)) {
```
The same for the other conditions, must be reviewed (these are 3 times in this PR).